packages/mdc-textfield/foundation.js

```diff
@@ -83,6 +100,16 @@ export default class MDCTextfieldFoundation extends MDCFoundation {
     this.showHelptext_();
   }
 
+  autoCompleteFocus_(keyInteraction) {
+    if (!keyInteraction) {
+      this.activateFocus_();
+    }
+  }
+
+  setKeyInteraction_() {
+    this.keyInteractionOccurred = true;
```

Comment:
I think you're right. What if we make `deactivateFocus()_` reset `keyInteractionOccurred` if the input value is an empty string. We already check for it in the method and I think it makes sense to add that to the list of concerns for `deactivateFocus()_`, and it solves for the above scenario.